azurelinuxagent/ga/update.py

```diff
@@ -1017,6 +1023,49 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    def _add_accept_tcp_firewall_rule_if_not_enabled(self, dst_ip):
+        # Helper to execute a run command, returns True if no exception
+        # Here we primarily check if an  iptable rule exist. True if it exits , false if not
+        def _execute_run_command(command):
+            try:
+                shellutil.run_command(command)
+                return True
+            except CommandError as e:
+                # return code 1 is expected while using the check command. Raise if encounter any other return code
+                if e.returncode != 1:
+                    raise
+            return False
+
+        try:
+            wait = self.osutil.get_firewall_will_wait()
+
+            # "-C" checks if the iptable rule is available in the chain. It throws an exception with return code 1 if the ip table rule doesnt exist
+            drop_rule = get_firewall_drop_command(wait, AddFirewallRules.CHECK_COMMAND, dst_ip)
+            if not _execute_run_command(drop_rule):
+                # DROP command doesn't exist indicates then none of the firewall rules are set yet
+                # exiting here as the environment thread will set up all firewall rules
+                return
```

Comment:
Maybe add an info log here to be clear why we didnt do something, would be helpful while debugging

Reply:
Yes, added a log statement